packages/node/src/trpc.ts

```diff
@@ -0,0 +1,44 @@
+import { getCurrentHub } from '@sentry/core';
+import { normalize } from '@sentry/utils';
+
+interface SentryTrpcMiddlewareOptions {
+  attachRpcInput?: boolean;
+}
+
+interface TrpcMiddlewareArguments<T> {
+  path: string;
+  type: 'query' | 'mutation' | 'subscription';
+  next: () => T;
+  rawInput: unknown;
+}
+
+/**
+ * Sentry tRPC middleware that names the handling transaction after the called procedure.
+ *
+ * Use the Sentry tRPC middleware in combination with the Sentry server integration,
+ * e.g. Express Request Handlers or Next.js SDK.
+ */
+export async function sentryTrpcMiddleware(options: SentryTrpcMiddlewareOptions = {}) {
+  return function <T>({ path, type, next, rawInput }: TrpcMiddlewareArguments<T>): T {
+    const hub = getCurrentHub();
+    const clientOptions = hub.getClient()?.getOptions();
+    const sentryTransaction = hub.getScope()?.getTransaction();
+
+    if (sentryTransaction) {
+      sentryTransaction.setName(`${path}()`, 'route');
+      sentryTransaction.op = 'rpc.server';
+
+      const trpcData: Record<string, unknown> = {
+        procedureType: type,
+      };
+
+      if (options.attachRpcInput !== undefined ? options.attachRpcInput : clientOptions?.sendDefaultPii) {
+        trpcData.procedureInput = normalize(rawInput);
+      }
+
+      sentryTransaction.setData('trpc', trpcData);
+    }
+
+    return next();
```

Comment:
You might want to log something if the procedure actually fails? https://trpc.io/docs/server/middlewares#logging

Reply:
Good point, but for now I don't think we need to add this. People will still need to use this integration with a proper server framework integration like the Express integration for it to work, and those integrations will handle the errors. 

We could at some point add an option that allows capturing of non-200 responses.